frontend/src/pages/RegisterInvestor.test.tsx

```diff
@@ -0,0 +1,82 @@
+import React from "react";
+import { render, screen, fireEvent } from "@testing-library/react";
+import RegisterInvestor from "./RegisterInvestor";
+
+const mockFetch = jest.fn((url, options) => {
+  const body = options && typeof options === "object" ? JSON.parse((options as any).body) : {};
```

Comment:
The test's JSON.parse((options as any).body) will throw if options.body is undefined or already an object. Parse safely by checking for a body string and handling already-parsed objects.

```
const body = options?.body
  ? (typeof options.body === "string" ? JSON.parse(options.body) : options.body)
  : {};
```